src/core.rs

```diff
@@ -604,8 +606,17 @@ impl<T> Webauthn<T> {
         // Token Binding ID for the connection.
 
         // Verify that the rpIdHash in authData is the SHA-256 hash of the RP ID expected by the Relying Party.
-        if data.authenticator_data.rp_id_hash != self.rp_id_hash {
-            return Err(WebauthnError::InvalidRPIDHash);
+        // Note that if we have an appid stored in the state and the client indicates it has used the appid extension,
+        // we check the hash against this appid instead of the Relying Party
+        let has_appid_enabled = rsp.extensions.as_ref().map(|e| e.appid).unwrap_or(false);
+        if has_appid_enabled && appid.is_some() {
+            if data.authenticator_data.rp_id_hash != compute_sha256(appid.as_ref().unwrap().as_bytes()) {
```

Comment:
can we match here? Checking `is_some` and then `unwrap`ing seems like an anti-pattern to me.